### DISCUSSION TITLE
How can I deploy/install PaddleOCR(paddlepaddle) on an AWS EC2 instance?

### DISCUSSION THREAD

**Discussion by Bobivankenobi** [PaddlePaddle/PaddleOCR#11018] [Q&A] [answered]
In my attempt to install PaddleOCR(paddlepaddle) on an Amazon EC2 instance running Amazon Linux 2023, I encountered challenges related to shared libraries and dependencies versions that ultimately prevented successful installation and deployment.

What is recommended OS for the EC2 instance (e.g., Ubuntu, Amazon Linux)?
Is this best practise when is about deploying such a software?
Can someone help me with this please. 🙏 

While the installation of paddlepaddle worked smoothly on my local environment (everything works fine) I encountered numerous problems when attempting to install it on an Amazon Linux 2023 AMI.

My small repo:
[https://github.com/Bobivankenobi/python-ocr/](https://github.com/Bobivankenobi/python-ocr/)

**Comment by HolonUniverse** [comment, marked as answer]
That's a great question.  Here's what I did to get this working:
1) I'm using the latest AL2023 - Amazon Linux 2023.4.20240429
2) Installed Python 3.11 using this tutorial: https://plainenglish.io/community/how-to-install-python-3-11-with-pip-on-amazon-linux-2023-9ab2ed
3) Used Python3.11 to create a virtual env and installed the requirements.txt that you provided in GitHub
4) To resolve the missing 'libgl.so.1' file, I executed 'sudo dnf install libglvnd-glx' which installed quite a few dependencies.
5) I updated line 64 of the app.py file to force Flask to run on the IP of the instance and a different port.

After that, I ran your python file:
$ python3.11 ./app.py

and my POST requests work like a champ.


Hope this helps and thanks for the app.py!
